src/t_string.c

```diff
@@ -106,14 +114,19 @@ void setCommand(client *c) {
 
         if ((a[0] == 'n' || a[0] == 'N') &&
             (a[1] == 'x' || a[1] == 'X') && a[2] == '\0' &&
-            !(flags & OBJ_SET_XX))
+            !(flags & OBJ_SET_XX) && !(flags & OBJ_SET_GET))
```

Comment:
i think the right thing would be **not** to avoid setting the flag, and then check for conflicting types at the end and responding with syntax error.

Reply:
The `nx`, `xx` and `ex` parameter uses the same approach of not setting the flag:
https://github.com/redis/redis/blob/unstable/src/t_string.c#L109
https://github.com/redis/redis/blob/unstable/src/t_string.c#L114
https://github.com/redis/redis/blob/unstable/src/t_string.c#L123

Also using the `GET` parameter with `NX` parameter will result in a syntax error, as tested by this test: https://github.com/redis/redis/pull/7852/files#diff-b8aa159a8e8eaff88807628299c63f05R408

Reply:
ohh, i get it. by not entering this `if` it reaches the final `else` and returns an error.
maybe it's indeed better than setting multiple flags and then checking for collisions...
maybe not... but it's already like that and it works, so let's keep it.